### PR TITLE
ARROW-17192: [Python] Pass **kwargs in read_feather to to_pandas()

### PR DESCRIPTION
**Pull request by AlenkaF**
Pass `**kwargs` in `read_feather` to `to_pandas()` to ensure `timestamp_as_object=True` (together with other `**kwargs`) can be passed and the conversion between Arrow and pandas doesn't fail due to different datetime resolutions.

### CODE REVIEW

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
@AlenkaF feel free to merge once CI is fixed and you can update here to get green CI

**Comment by AlenkaF** [comment]
The release jobs keep failing even after rebasing, the MacOs failure has a Jira: https://issues.apache.org/jira/browse/ARROW-18150.

None of these failures seem to be connected to this PR.

File: python/pyarrow/tests/test_feather.py [hunk 1]
```diff
@@ -838,3 +838,26 @@ def test_preserve_index_pandas(version):
...
+    # ARROW-17192 - ensure timestamp_as_object=True (together with other
+    # **kwargs) can be passed in read_feather to to_pandas.
+
+    # file generated with:
+    #   from datetime import datetime
+    #   df = pd.DataFrame({"date": [
+    #       datetime.fromisoformat("1654-01-01"),
+    #       datetime.fromisoformat("1920-01-01"), ],
+    #   })
+    #   df.to_feather(datadir / "test_resolution.feather")
```
Comment (jorisvandenbossche) [inline python/pyarrow/tests/test_feather.py:854]: Is there a reason to not do this on the fly in the test in a tempdir?

**Comment by jorisvandenbossche** [comment]
Updated test looks good! 
There is still the data file committed though, that can be removed now

**Comment by AlenkaF** [comment]
OK, this should be ready now. Can merge when the CI is green.

**Comment by adrienpacifico** [comment]
Thank you very much @AlenkaF !

**Pull request merged**